### DISCUSSION TITLE
Failure to build - BLTouch-3x3-HighSpeed

### DISCUSSION THREAD

**Discussion by Nuutyl** [Jyers/Marlin#1939] [Q&A] [answered]
I was trying to get BLTOUCH_HS_MODE to work on my previous build by uncommenting the HS_MODE enable line, but ran into an error when building in VSCode with Auto Build Marlin. 
I tried building the BLTouch-3x3-HighSpeed (after enabling FAN_SOFT_PWM) and got the same error. 

The errors are pointing to this line:
#if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0

Copy-paste the relevant terminal output:
`In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\HAL.cpp:30:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\HAL_SPI.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\MarlinSerial.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\Servo.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\eeprom_bl24cxx.cpp:32:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\eeprom_flash.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\eeprom_if_iic.cpp:32:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\eeprom_sdcard.cpp:31:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\eeprom_sram.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\eeprom_wired.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:49,
                 from Marlin\src\HAL\STM32\fast_pwm.cpp:27:
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:1586:53: error: operator '&&' has no right operand
 1586 |     #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      |                                                     ^~
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\HAL_SPI.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\MarlinSerial.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\Servo.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\eeprom_flash.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\HAL.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\eeprom_bl24cxx.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\eeprom_sdcard.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\eeprom_wired.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\eeprom_sram.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality\src\src\HAL\STM32\fast_pwm.cpp.o] Error 1`

I tried looking around for other lines referencing BLTOUCH_HS_MODE in config and config_adv, but no luck. 

What should I try next?

**Comment by Nuutyl** [comment, marked as answer]
Narrowed it down to this if statement in SanityCheck.h 

 #if ENABLED(BLTOUCH_HS_MODE) && BLTOUCH_HS_MODE == 0
      #error "BLTOUCH_HS_MODE must now be defined as true or false, indicating the default state."
    #endif

What am I missing with this function? if BLTOUCH_HS_MODE is enabled AND it equals zero, throw an error? What's wrong with the format that the && isn't picking up on BLTOUCH_HS_MODE == 0?


EDIT:
As @adam3654 mentioned below, ensure the BLTOUCH_HS_MODE has a true following it.

#define BLTOUCH_HS_MODE true

**Reply by adam3654** [reply]
Check to make sure that the #define has `true` after it in the config file

**Reply by Nuutyl** [reply]
That did it. Thanks Adam!
